# Superstore Sales & Profit Analysis

This project analyzes sales and profitability data from a retail superstore to
understand how different product categories and sub-categories contribute to
overall business performance. The main objective is to identify categories that
generate high sales but suffer from low profitability, and to uncover the root
causes behind these patterns.

The analysis focuses on sales, profit, profit margin, discount behavior, and
transaction-level performance to support data-driven business recommendations.

Proyek ini menganalisis data penjualan dan profit dari sebuah retail superstore
untuk memahami bagaimana setiap kategori dan sub-category produk berkontribusi
terhadap performa bisnis secara keseluruhan. Tujuan utama analisis ini adalah
mengidentifikasi kategori dengan penjualan tinggi namun profitabilitas rendah,
serta menemukan akar permasalahan di balik kondisi tersebut.

Analisis difokuskan pada penjualan, profit, profit margin, pola diskon, dan data
transaksi untuk mendukung rekomendasi bisnis berbasis data.




---



## Problem Statement & Key Questions

Despite strong overall sales, the business experiences uneven profitability
across product categories. Some categories appear to perform well in terms of
sales volume, yet contribute relatively little to total profit. This raises
important questions about pricing efficiency and discount strategies.

The key business questions addressed in this analysis are:
- Which product categories generate the highest sales and profit?
- Are there categories with high sales but low profit margins?
- What factors contribute to low profitability in those categories?
- Which sub-categories are the primary drivers of losses?
- How do discount strategies impact profit performance?

Meskipun penjualan secara keseluruhan tergolong tinggi, bisnis ini menunjukkan
ketimpangan profitabilitas antar kategori produk. Beberapa kategori terlihat
memiliki penjualan yang baik, namun kontribusinya terhadap profit relatif kecil.
Hal ini menimbulkan pertanyaan mengenai efisiensi harga dan strategi diskon.

Pertanyaan bisnis utama yang dijawab dalam analisis ini meliputi:
- Kategori produk mana yang menghasilkan penjualan dan profit tertinggi?
- Apakah terdapat kategori dengan penjualan tinggi namun profit margin rendah?
- Faktor apa yang menyebabkan rendahnya profitabilitas pada kategori tersebut?
- Sub-category mana yang menjadi penyumbang utama kerugian?
- Bagaimana pengaruh strategi diskon terhadap performa profit?




---



## Dataset Overview & Analysis Approach

The dataset used in this analysis is the Superstore Sales dataset obtained from
Kaggle. It contains transaction-level data, including order information,
customer details, product categories, sales, profit, quantity, and discount
values.

Before conducting the analysis, basic data inspection was performed to ensure
data consistency and readiness for analysis. The analytical approach follows
a structured workflow:
1. Analyze total sales and profit by category.
2. Evaluate profit margin to identify efficiency issues.
3. Investigate discount patterns across categories.
4. Drill down into sub-category performance, particularly within Furniture.
5. Analyze transaction-level data to identify root causes of losses.
6. Formulate data-driven business recommendations.

Dataset yang digunakan dalam analisis ini adalah Superstore Sales Dataset yang
diperoleh dari Kaggle. Dataset ini berisi data transaksi, termasuk informasi
pesanan, detail pelanggan, kategori produk, penjualan, profit, kuantitas, dan
diskon.

Sebelum analisis dilakukan, pengecekan data dasar dilakukan untuk memastikan
konsistensi dan kesiapan data. Pendekatan analisis dilakukan secara terstruktur
dengan alur sebagai berikut:
1. Menganalisis total penjualan dan profit per kategori.
2. Menghitung profit margin untuk mengidentifikasi efisiensi.
3. Menganalisis pola diskon pada setiap kategori.
4. Melakukan drill-down ke level sub-category, khususnya pada Furniture.
5. Menganalisis data transaksi untuk menemukan akar permasalahan kerugian.
6. Menyusun rekomendasi bisnis berbasis data.




---



## 1. Load Dataset

In this section, the Superstore dataset is loaded into Python using the pandas library.

Pada bagian ini, dataset Superstore dimuat ke dalam Python menggunakan library pandas.

In [2]:
import pandas as pd

df = pd.read_csv('Sample - Superstore.csv', encoding='latin-1')
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164




---



## 2. Data Overview

We examine the structure of the dataset, including the number of rows, columns,
and data types. This step ensures that the dataset is ready for further analysis.

Pada tahap ini, dilakukan pengecekan struktur dataset, termasuk jumlah baris,
jumlah kolom, dan tipe data. Langkah ini bertujuan untuk memastikan bahwa
dataset siap digunakan untuk analisis lanjutan.


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity



---



## Analysis Results

This section presents the results of the analysis, starting from high-level
category performance and gradually drilling down into sub-category and
transaction-level insights to identify the root causes of profitability issues.

Pada bagian ini disajikan hasil analisis yang dimulai dari performa kategori
secara umum, kemudian dilanjutkan dengan analisis yang lebih mendalam hingga
ke level sub-category dan transaksi untuk mengidentifikasi akar permasalahan
profitabilitas.


## Profit by Product Category

This analysis aims to identify which product categories contribute the most
to overall business profitability. By grouping data by category and summing
the profit, we can evaluate business performance beyond sales volume.

Analisis ini bertujuan untuk mengetahui kategori produk mana yang memberikan
kontribusi keuntungan terbesar bagi bisnis. Dengan mengelompokkan data
berdasarkan kategori dan menjumlahkan profit, kita dapat menilai performa
bisnis secara lebih akurat, bukan hanya dari sisi penjualan.


In [4]:
profit_by_category = df.groupby('Category')['Profit'].sum()
profit_by_category

,Profit
Category,
Furniture,18451.2728
Office Supplies,122490.8008
Technology,145454.9481




---



## Sales by Product Category

This section analyzes total sales for each product category.
The result is later compared with profit data to identify categories
with high sales but relatively low profitability.

Pada bagian ini, dilakukan analisis total penjualan untuk setiap kategori produk.
Hasil ini kemudian dibandingkan dengan data profit untuk mengidentifikasi
kategori yang memiliki penjualan tinggi tetapi keuntungan yang tidak sebanding.


In [5]:
sales_by_category = df.groupby('Category')['Sales'].sum()
sales_by_category

,Sales
Category,
Furniture,741999.7953
Office Supplies,719047.0320
Technology,836154.0330




---



## Category Performance Analysis: Sales, Profit, and Profit Margin

In this section, we combine total sales and total profit for each product category
into a single table. We then calculate the profit margin to better understand
how efficiently each category generates profit relative to its sales.

This analysis helps identify categories that may appear strong based on sales
performance but are actually less profitable when profitability efficiency
is taken into account.

Pada bagian ini, data total penjualan (sales) dan total keuntungan (profit)
untuk setiap kategori digabungkan ke dalam satu tabel. Selanjutnya, dilakukan
perhitungan profit margin untuk melihat seberapa efisien setiap kategori dalam
menghasilkan keuntungan dibandingkan dengan nilai penjualannya.

Analisis ini bertujuan untuk mengidentifikasi kategori yang terlihat kuat dari
sisi penjualan, namun sebenarnya kurang menguntungkan dari sisi profitabilitas.


In [6]:
category_performance = pd.DataFrame({
    'Sales': sales_by_category,
    'Profit': profit_by_category
})

category_performance['Profit Margin (%)'] = (category_performance['Profit'] / category_performance['Sales']) * 100

# Sort by sales in descending order and profit margin in ascending order to easily spot high sales, low profit
category_performance_sorted = category_performance.sort_values(by=['Sales', 'Profit Margin (%)'], ascending=[False, True])

display(category_performance_sorted)

,Sales,Profit,Profit Margin (%)
Category,,,
Technology,836154.0330,145454.9481,17.395712
Furniture,741999.7953,18451.2728,2.486695
Office Supplies,719047.0320,122490.8008,17.035158




---



## Average Discount by Category

In this section, we analyze the average discount applied to each product category.
The goal is to verify whether categories with lower profit margins are associated
with higher discount levels.

This analysis helps validate whether discounting strategy is a key factor
contributing to low profitability.

Pada bagian ini, dilakukan analisis rata-rata diskon untuk setiap kategori produk.
Tujuannya adalah untuk membuktikan apakah kategori dengan profit margin rendah
memang memiliki tingkat diskon yang lebih tinggi.

Analisis ini digunakan untuk menguji apakah strategi pemberian diskon menjadi
faktor utama penyebab rendahnya profitabilitas.


In [9]:
avg_discount_by_category = df.groupby('Category')['Discount'].mean()
avg_discount_by_category


,Discount
Category,
Furniture,0.173923
Office Supplies,0.157285
Technology,0.132323




---



## Furniture Sub-Category Profit Analysis

In this section, we drill down into the Furniture category to analyze profit
performance at the sub-category level. The objective is to identify which
Furniture sub-categories contribute the most to losses or low profitability.

This deeper analysis helps pinpoint specific problem areas within the Furniture
category that may require changes in pricing, discount strategy, or inventory
management.

Pada bagian ini, analisis difokuskan pada kategori Furniture hingga ke level
sub-category untuk melihat performa profit secara lebih detail. Tujuannya adalah
mengidentifikasi sub-category Furniture yang paling berkontribusi terhadap
rendahnya profit atau bahkan mengalami kerugian.

Analisis ini membantu menentukan area spesifik yang bermasalah dan dapat menjadi
dasar pengambilan keputusan bisnis, seperti penyesuaian diskon, harga, atau jumlah
stok.


In [10]:
# Filter data to Furniture category only
furniture_df = df[df['Category'] == 'Furniture']

# Calculate total profit per Furniture sub-category
profit_by_furniture_subcat = (
    furniture_df
    .groupby('Sub-Category')['Profit']
    .sum()
    .sort_values()
)

profit_by_furniture_subcat

,Profit
Sub-Category,
Tables,-17725.4811
Bookcases,-3472.5560
Furnishings,13059.1436
Chairs,26590.1663




---



## Discount vs Profit Analysis for Tables Sub-Category

In this section, we focus on the Tables sub-category to examine the relationship
between discount levels and profit. Since Tables have been identified as the
largest contributor to losses within the Furniture category, this analysis aims
to validate whether high discounting directly correlates with negative profit.

By analyzing discount and profit together at the transaction level, we can
confirm whether aggressive discount strategies are the primary driver of losses.

Pada bagian ini, analisis difokuskan pada sub-category Tables untuk melihat
hubungan antara tingkat diskon dan profit. Karena Tables telah terbukti menjadi
penyumbang kerugian terbesar dalam kategori Furniture, analisis ini bertujuan
untuk membuktikan apakah diskon yang tinggi secara langsung berkorelasi dengan
profit negatif.

Analisis ini dilakukan pada level transaksi untuk memastikan bahwa strategi
diskon yang agresif memang menjadi penyebab utama kerugian.


In [11]:
# Filter data to Tables sub-category only
tables_df = df[
    (df['Category'] == 'Furniture') &
    (df['Sub-Category'] == 'Tables')
]

# Descriptive statistics for Discount and Profit
tables_df[['Discount', 'Profit']].describe()

,Discount,Profit
count,319.000000,319.000000
mean,0.261285,-55.565771
std,0.169622,230.565006
min,0.000000,-1862.312400
25%,0.200000,-122.829300
50%,0.300000,-31.372200
75%,0.400000,33.851700
max,0.500000,629.010000




---



## Final Conclusion

This analysis reveals that high sales performance does not necessarily translate
into high profitability. Among all product categories, Furniture stands out as
a category with relatively high sales but significantly low profit margin.

A deeper analysis shows that the primary source of this issue comes from the
Tables sub-category. Tables consistently generate negative profit, driven by
aggressive discounting strategies. Transaction-level analysis confirms that
most Tables sales involve high discounts (averaging around 26%), resulting in
systematically negative profit despite steady demand.

These findings indicate that the losses in Furniture are structural rather than
incidental, and are largely caused by pricing and discount policies rather than
lack of sales volume.

Analisis ini menunjukkan bahwa tingginya penjualan tidak selalu berarti tingginya
keuntungan. Di antara seluruh kategori produk, Furniture terlihat memiliki nilai
penjualan yang cukup besar, namun profit margin yang sangat rendah.

Analisis lebih lanjut mengungkap bahwa sumber utama permasalahan berasal dari
sub-category Tables. Tables secara konsisten menghasilkan profit negatif akibat
strategi diskon yang terlalu agresif. Analisis pada level transaksi membuktikan
bahwa sebagian besar penjualan Tables melibatkan diskon tinggi (rata-rata sekitar
26%), sehingga menyebabkan kerugian yang bersifat sistemik meskipun permintaan
pasar cukup stabil.

Temuan ini menunjukkan bahwa kerugian pada kategori Furniture bukan disebabkan oleh
rendahnya penjualan, melainkan oleh kebijakan harga dan diskon yang kurang optimal.


## Business Recommendations

Based on the analysis, it is recommended to optimize discount strategies for the
Furniture category, particularly for the Tables sub-category. Rather than relying
on aggressive discounting to drive sales, the business should evaluate more
selective or gradual discount approaches to protect profit margins.

Possible actions include testing lower discount thresholds for Tables, limiting
high discounts to specific periods or customer segments, and closely monitoring
profit impact after discount adjustments. These measures can help improve overall
profitability without significantly reducing sales volume.

Berdasarkan hasil analisis, disarankan untuk melakukan optimasi strategi diskon
pada kategori Furniture, khususnya sub-category Tables. Alih-alih mengandalkan
diskon besar untuk mendorong penjualan, bisnis perlu menerapkan pendekatan diskon
yang lebih selektif atau bertahap guna menjaga margin keuntungan.

Beberapa langkah yang dapat dipertimbangkan antara lain menguji batas diskon yang
lebih rendah untuk produk Tables, membatasi diskon besar pada periode atau segmen
pelanggan tertentu, serta memantau dampak perubahan diskon terhadap profit secara
berkelanjutan. Pendekatan ini diharapkan dapat meningkatkan profitabilitas tanpa
mengorbankan volume penjualan secara signifikan.


## TL;DR — Key Insights Summary

- Technology is the most profitable category, driven by higher profit margins,
  even though Office Supplies and Furniture also generate substantial sales.
- Furniture appears strong in sales but performs poorly in profitability,
  with the lowest profit margin among all categories.
- Profit margin analysis reveals that Furniture efficiency is significantly
  lower compared to Technology and Office Supplies.
- Discount analysis confirms that Furniture has the highest average discount,
  which strongly correlates with low profitability.
- At the sub-category level, Tables are the primary source of losses within
  Furniture, showing consistently high discounts and negative profit per transaction.
- Optimizing discount strategies for Tables could significantly improve overall
  business profitability without reducing sales volume drastically.

Ringkasan singkat:
- Kategori Technology menghasilkan profit tertinggi.
- Furniture terlihat laku tetapi memiliki profit margin terendah.
- Diskon tinggi terbukti menjadi penyebab utama rendahnya profit Furniture.
- Sub-category Tables merupakan sumber kerugian terbesar.
- Rekomendasi utama adalah optimasi strategi diskon pada Tables.
